doc/internationalization.md

```diff
@@ -0,0 +1,44 @@
+gRPC Internationalization
+=========================
+
+As a universal RPC framework, gRPC needs to be fully usable within/across different international environments. 
+This document describes gRPC API and behavior specifics when used in a non-english environment.
+
+## API Concepts
+
+While some API elements need to be able to represent non-english content, some are intentionally left as ASCII-only
+for simplicity & performance reasons.
+
+### Method name (in RPC Invocation)
+Method names are ASCII-only. Most gRPC services will use protobuf which only allows ASCII based method names anyway.
+Also, handling method names is a very hot code path.
+
+Recommended representation in language-specific APIs: string type.
+
+### Host name (in RPC Invocation)
+Host names are punycode encoded. Currently, the punycode needs to be provided by the user.
+
+Recommended representation in language-specific APIs: string/unicode string.
```

Comment:
We're not doing it today, but it looks like it could be supported in Java via [IDN](http://docs.oracle.com/javase/8/docs/api/java/net/IDN.html). It's been supported for a while even on [Android](https://developer.android.com/reference/java/net/IDN.html).